solution/backend/file_manager/admin.py

```diff
@@ -65,10 +65,12 @@ def save_model(self, request, obj, form, change):
         if path:
             obj.file_name = self.clean_file_name(path._name)
             self.upload_file(path, obj)
-        form.instance.save()
-        add_to_index(form.instance)
         super().save_model(request, obj, form, change)
 
+    def save_related(self, request, form, formsets, change):
```

Comment:
save_related is needed to take care of the many to many relationships (locations, and subjects)

Reply:
It was previously not doing this because it was done in the normal save method.